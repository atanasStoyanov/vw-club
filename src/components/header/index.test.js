import React from 'react';
import renderer from 'react-test-renderer';
import Header from '.';
import TestingEnvironmet from '../../test-utils/router';

describe('Header Component', () => {
    it('should render authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironmet value={{
                loggedIn: true,
                user: {
                    id: '123'
                }
            }} >

                <Header />
            </TestingEnvironmet>
        ).toJSON();
        expect(tree).toMatchSnapshot();

    });

    it('should render non-authenticated routes', () => {
        const tree = renderer.create(
            <TestingEnvironmet value={{
                loggedIn: false,
            }} >

                <Header />
            </TestingEnvironmet>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
})

function test() {

}