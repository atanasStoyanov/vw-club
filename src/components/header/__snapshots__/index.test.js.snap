// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt="vw-logo"
    className="logo"
    src="vw-logo.png"
  />
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/forum"
      onClick={[Function]}
    >
      Forum
    </a>
  </div>
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/profile/123"
      onClick={[Function]}
    >
      Profile
    </a>
  </div>
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Logout
    </a>
  </div>
</header>
`;

exports[`Header Component should render non-authenticated routes 1`] = `
<header
  className="navigation"
>
  <img
    alt="vw-logo"
    className="logo"
    src="vw-logo.png"
  />
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
  </div>
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/login"
      onClick={[Function]}
    >
      Login
    </a>
  </div>
  <div
    className="header-list-item"
  >
    <a
      className="header-link"
      href="/register"
      onClick={[Function]}
    >
      Register
    </a>
  </div>
</header>
`;
